upload.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2014 Tamás Gulácsi. All rights reserved.
+// Use of this source code is governed by an Apache 2.0
+// license that can be found in the LICENSE file.
+
+package picago
+
+import (
+	"bytes"
+	"fmt"
+	"html"
+	"io"
+	"io/ioutil"
+	"mime/multipart"
+	"net/http"
+	"net/textproto"
+	"strconv"
+	"strings"
+)
+
+/*
+Upload photo
+If userID is empty, "default" is used.
+If albumID is empty, "default" is used.
+
+fileName is the image's filename
+summary is the caption of the photo.
+MIME is the Content-Type, only support "image/bmp", "image/gif", "image/jpeg", and "image/png"
+*/
+func UploadPhoto(client *http.Client, userID, albumID, fileName, summary, MIME string, photoRaw []byte) (*Photo, error) {
+	if userID == "" {
+		userID = "default"
+	}
+	if albumID == "" {
+		albumID = "default"
+	}
+	url := strings.Replace(photoURL, "{userID}", userID, 1)
+	url = strings.Replace(url, "{albumID}", albumID, 1)
+	url = url[0:strings.LastIndex(url, "?")]
+
+	buf := bytes.NewBuffer(nil)
+	buf.WriteString("Media multipart posting\r\n")
+	w := multipart.NewWriter(buf)
+	w.SetBoundary("_END_OF_PART_")
```

Comment:
Will remove ```buf.WriteString("Media multipart posting\r\n")```, it still worked in my test.
Is that mean do not need set boundary?